### PR TITLE
documentation: smdistributed libraries release notes

### PR DESCRIPTION
**Pull request by mchoi8739**
*Description of changes:*

smdistributed libraries release notes

*Testing done:*

tox sphinx passed

RTD PR builder passed:
- https://sagemaker--3543.org.readthedocs.build/en/3543/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.html#sagemaker-distributed-data-parallel-1-6-0-release-notes
- https://sagemaker--3543.org.readthedocs.build/en/3543/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.html#sagemaker-distributed-model-parallel-1-13-0-release-notes

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst [hunk 1]
```diff
@@ -7,9 +7,44 @@ Release Notes
...
 
-SageMaker Distributed Data Parallel 1.5.0 Release Notes
+SageMaker Distributed Data Parallel 1.6.0 Release Notes
 =======================================================
 
+*Date: Dec. 15. 2022*
+
+**New Features**
+
+* New SMDDP Collectives support for the SageMaker model parallelism library’s sharded data parallelism operating AllGather. For more information, see `Sharded data parallelism with SMDDP Collectives <https://docs.aws.amazon.com/sagemaker/latest/dg/model-parallel-extended-features-pytorch-sharded-data-parallelism.html#model-parallel-extended-features-pytorch-sharded-data-parallelism-smddp-collectives>`_ in the Amazon SageMaker Developer Guide.
```
Comment (Zha0q1) [inline doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst:17]: nit:
"Optimized implementation of AllGather in the SMDDP Collective package to accelerate shared data parallelism training for the SageMaker Model Parallelism Library."

File: doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst [hunk 2]
```diff
@@ -7,9 +7,47 @@ Release Notes
...
+* Added support for Amazon EC2 ml.p4de.24xlarge instances.
```
Comment (Arjunbala) [inline doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst:18]: Let's clarify this is only for All-Reduce

File: doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst [hunk 3]
```diff
@@ -7,9 +7,47 @@ Release Notes
...
 
-SageMaker Distributed Data Parallel 1.5.0 Release Notes
+SageMaker Distributed Data Parallel 1.6.0 Release Notes
 =======================================================
 
+*Date: Dec. 15. 2022*
+
+**New Features**
+
+* New SMDDP Collectives support for the SageMaker model parallelism library’s sharded data parallelism operating AllGather. For more information, see `Sharded data parallelism with SMDDP Collectives <https://docs.aws.amazon.com/sagemaker/latest/dg/model-parallel-extended-features-pytorch-sharded-data-parallelism.html#model-parallel-extended-features-pytorch-sharded-data-parallelism-smddp-collectives>`_ in the Amazon SageMaker Developer Guide.
```
Comment (Arjunbala) [inline doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst:17]: Sentence structuring seems off to me. Would suggest to rewrite as - "New SMDDP AllGather collective to complement the sharded data parallelism technique in SageMaker model parallelism library"

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst [hunk 4]
```diff
@@ -6,9 +6,61 @@ New features, bug fixes, and improvements are regularly made to the SageMaker
...
 
-SageMaker Distributed Model Parallel 1.11.0 Release Notes
+SageMaker Distributed Model Parallel 1.13.0 Release Notes
 =========================================================
 
+*Date: Dec. 15. 2022*
+
+**New Features**
+
+* Sharded data parallelism now supports a new backend for collectives, SMDDP. For supported scenarios
```
Comment (Zha0q1) [inline doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst:16]: nit:
"Sharded data parallelism now supports a new, more optimized process group backend option, "smdp," for collective calls. Under supported training configurations, "smddp" backend is enabled by default.

**Review by Zha0q1 — APPROVED** [review]
LGTM, thanks Miyoung!

File: doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst [hunk 5]
```diff
@@ -7,9 +7,47 @@ Release Notes
...
+* Added support for Amazon EC2 ml.p4de.24xlarge instances.
+
+**Improvements**
+
+* Improved general performance of the SMDDP AllReduce collective communication operation.
```
Comment (carolynwang) [inline doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst:22]: wanna check the wording - "general performance" vs "general improvements" is different, @Arjunbala?

**Review by carolynwang — APPROVED** [review]

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst (hunk 4 above)
Comment (mchoi8739) [reply doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst:16]: I don't see that "smdp" is an option exposed externally, neither "smddp"

File: doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst (hunk 2 above)
Comment (mchoi8739) [reply doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst:18]: Pardon, what do you mean? I think AllReduce is not released

**Review by knikure — APPROVED** [review]

File: doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst (hunk 2 above)
Comment (Zha0q1) [reply doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst:18]: only allreduce/ddp training works on p4de at the moment.
Allgather only works on p4 and is not yet p4de ready

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst (hunk 4 above)
Comment (Zha0q1) [reply doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst:16]: Makes sense, from users' perspective they only see "nccl" or "auto". How about

"Sharded data parallelism is now accelerated by SMDDP Collectives. Under supported training configurations, "smddp" is automatically selected as the process group backend.

Comment (mchoi8739) [reply doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst:16]: incorporating using Arjun's suggestion

File: doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst (hunk 5 above)
Comment (mchoi8739) [reply doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst:22]: Arjun first suggested "General performance improvements to SMDDP All-Reduce collective". Let me know if we just want to keep it as he suggested.

Comment (mchoi8739) [reply doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst:22]: I'm going to just directly use Arjun's suggestion right now.

Comment (Arjunbala) [reply doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.rst]: It's both, so we're good :)

**Pull request merged**